app/controllers/student_applications_controller.rb

```diff
@@ -0,0 +1,29 @@
+class StudentApplicationsController < ApplicationController
+  before_action :authenticate_applicant!
+  before_action :unique_semester_application
+
+  def new
+    @student_application = current_user.student_applications.build
+  end
+
+  def create
+    @student_application = current_user.student_applications.build student_application_params
+    @student_application.semester = Semester.current_semester
```

Comment:
If you write `include AssignToCurrentSemester` in the model then this will automatically happen and you don't have to write this line :)